src/main/webapp/app/entities/submission.model.ts

```diff
@@ -40,3 +40,58 @@ export abstract class Submission implements BaseEntity {
         this.submitted = false; // default value
     }
 }
+
+/**
+ * Used to access the correct submissions result
+ *
+ * @param submission
+ */
+export function getLatestSubmissionResult(submission: Submission | undefined): Result | undefined {
+    checkForResultsLength(submission);
+    if (submission?.results) {
+        const length = submission.results.length;
+        if (length > 0) {
+            return submission.results[length - 1];
+        }
+    }
+}
+
+/**
+ * Used to set the correct submissions result
```

Comment:
What is the correct submissions result?

Reply:
I think this needs a little more elaboration :)

Reply:
This should also be fixed now.